service/service.go

```diff
@@ -178,19 +168,34 @@ func New(config Config) (*Service, error) {
 
 	var clusterController *controller.Cluster
 	{
+		// These credentials will be used when creating AzureClients for Control Plane clusters.
+		gsClientCredentialsConfig, err := credential.NewAzureCredentials(
+			config.Viper.GetString(config.Flag.Service.Azure.ClientID),
+			config.Viper.GetString(config.Flag.Service.Azure.ClientSecret),
+			config.Viper.GetString(config.Flag.Service.Azure.TenantID),
```

Comment:
If I understood correctly, in `credential.NewAzureCredentials(...)` the credentials passed in from CLI arguments will have precedence over environment variables, right?
In that case I would maybe make it more explicit by moving these 3 lines that are reading the flags to be inside of the function. Not a blocker :)

Reply:
I didn't want to make the credential package to depend on Viper or whatever we use for the CLI parameters. Also, I believe by reading the code as it is now, it looks like you are getting credentials for those parameters values, and that's what we are doing. It's just that if the rare case that they are empty, we will try to use env vars, but you as a client don't need to care because you are passing the parameters. Does that make sense?

Reply:
Sounds good!